electron/src/lib/CertificateVerifyProcManager.ts

```diff
@@ -0,0 +1,184 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import * as certificateUtils from '@wireapp/certificate-check';
+import {app, dialog} from 'electron';
+import * as path from 'path';
+import {URL} from 'url';
+import * as environment from '../js/environment';
+import {getText} from '../locale/locale';
+import {LogFactory} from '../util/';
+
+const LOG_DIR = path.join(app.getPath('userData'), 'logs');
+LogFactory.LOG_FILE_PATH = LOG_DIR;
+LogFactory.LOG_FILE_NAME = 'electron.log';
+
+const logger = LogFactory.getLogger('CertificateVerifyProcManager.ts', {forceEnable: true});
+
+interface DisplayCertificateErrorOptions {
+  bypassDialogLock: boolean;
+  isChromiumError: boolean;
+  isCheckboxChecked: boolean;
+}
+
+class CertificateVerifyProcManager {
+  private static bypassCertificatePinning: boolean = false;
+  private static isDialogLocked: boolean = false;
+
+  public static readonly CHROMIUM_ERRORS = {
+    CERT_AUTHORITY_INVALID: -202,
+    CERT_COMMON_NAME_INVALID: -200,
+  };
+  private static readonly RESPONSE = {
+    RETRY: 0,
+    SHOW_DETAILS: 1,
+  };
+
+  private static displayCertificateDetails(
+    hostname: string,
+    certificate: Electron.Certificate,
+    options: DisplayCertificateErrorOptions
+  ) {
+    dialog.showCertificateTrustDialog(
+      {
+        certificate,
+        message: `${
+          options.isChromiumError
+            ? getText('certificateVerifyProcManagerShowDetailsTextChromium')
+            : getText('certificateVerifyProcManagerShowDetailsTextPinning')
+        } ${hostname}`,
```

Comment:
Please create a few constants here instead of moving it all into one line. This is not well readable.